presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -5916,6 +5916,9 @@ public void testInformationSchemaFiltering()
         assertQuery(
                 "SELECT table_name FROM information_schema.tables WHERE table_name = 'orders' LIMIT 1",
                 "SELECT 'orders' table_name");
+        assertQuery(
```

Comment:
I don't understand why this passes (or I assume it does... travis hasn't finished).  Before we were seeing the predicates in this query continually reordered in calls to PickTableLayout, which is why I had to add https://github.com/prestodb/presto/pull/8735/files#diff-170a0e9779dbbc113341e859216cc24fR113.  I don't think objects.equals(), which is how we check for predicate equivalence now should take care of that.  But I guess so long as it works it's fine?

Reply:
> query continually reordered in calls to PickTableLayout

To me it is kind of a bug, that would mean planner is not deterministic.

UPDATE:
Somehow when I was testing this test it was OK. Maybe I was on wrong commit, or forgot to run after I removed `ExpressionEquivalence`. You were right, this test looped `PickTableLayout`. Collection.unmodifiable* used in `DomainTranslator` is non deterministic. I fixed that.